app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/hplus/HPlusHandlerThread.java

```diff
@@ -391,14 +391,15 @@ public boolean processRealtimeStats(byte[] data) {
             sample.setRawHPlusHealthData(record.getRawData());
             sample.setProvider(provider);
 
+            provider.addGBActivitySample(sample);
+
             sample.setSteps(sample.getSteps() - prevRealTimeRecord.steps);
```

Comment:
I think the steps will not be committed to the db when set *after* addGBActivitySample(). If that is intended, I suggest to add a short comment.